API.md

```diff
@@ -0,0 +1,203 @@
+# CodeReady Analytics Core API Specifications
+
+CodeReady Analytics is an open-source analytics platform aimed primarily at assisting developers in getting insights and recommendations on the selected dependency/package stack used for developing their applications.
+
+This documentation is for developers who are interested in integrating their services or applications with CodeReady Analytics platform.
+
+## **Supported ecosystems**
+
+Below are the ecosystems that CodeReady Analytics currently supports.  We will be adding support for more ecosystems in the near future.
+| | |
+|-|-|
+|**Ecosystem**|**Manifest file**|
+| Java Maven | pom.xml |
+| Python Pypi | requirements.txt |
+| NodeJS | package.json |
+
+## **Prerequisites**
+
+Any new integration with the CodeReady Analytics platform will have to do that using an existing 3-scale API management gateway setup. Please reach out to us by raising an issue [here](https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension/issues)
+
+## **Initiate Dependency Analysis**
+
+Submits a dependency stack analysis request
```

Comment:
Can we use the format used in this doc here: https://docs.openshift.io/ct-analytics-integration-api-guide.html#ct_api_post_stack_analyses_request

POST /stack-analyses

Purpose: 

Here is the asciidoc source file for it: https://raw.githubusercontent.com/fabric8io/fabric8-online-docs/master/docs/topics/modules/api_post_stack_analyses_request.adoc
Here it is unclear what 'Submits a dependency...'

Reply:
We wanted to avoid the usage of stack report initially and rather use the word dependency to make it more understandable for external users, hence the change.

Reply:
updated purpose as per your suggestion